### PR TITLE
Auto logout in case of bad authentication

### PR DESCRIPTION
**Pull request by FrancoisPog**

### CODE REVIEW

**Review by valentinperignon — APPROVED** [review]
Great

File: client/src/App.js [hunk 1]
```diff
@@ -44,16 +44,22 @@ axios.interceptors.response.use(
   (response) => response,
   async (error) => {
     const originalRequest = error.config;
+    if (originalRequest.url === "/api/v1/users/logout") {
+      // to avoid loop
+      return Promise.reject(error);
+    }
+
     let { refreshToken } = AuthService.getCurrentUser() || {};
```
Comment (nathanaelhoun) [inline client/src/App.js:52]: Not related with your changes but still

```suggestion
    const { refreshToken } = AuthService.getCurrentUser() || {};
```

File: client/src/services/auth.service.js [hunk 2]
```diff
@@ -59,9 +59,11 @@ async function logout() {
     await axios.post("/api/v1/users/logout", { refreshToken });
   } catch {
     // Ignore error
+  } finally {
+    localStorage.removeItem(LOCAL_STORAGE_KEY);
+    queryClient.clear();
+    window.location.replace("/login");
```
Comment (nathanaelhoun) [inline client/src/services/auth.service.js:65]: If doing this here, we should 
- make it clear in the function name that we reload the document.
- remove the call to `location.replace` after `AuthService.logout()` in `Profile.js`
- make sure that no code is executed after `AuthService.logout()` in the whole app

**Review by nathanaelhoun — CHANGES_REQUESTED** [review]

File: client/src/services/auth.service.js (hunk 2 above)
Comment (FrancoisPog) [reply client/src/services/auth.service.js:65]: You're right is not the place to do that, I'll change that !

**Review by nathanaelhoun — APPROVED** [review]
Perfect!

**Pull request merged**